sdk/src/common/core.cc

```diff
@@ -7,8 +7,13 @@
 #include "opentelemetry/nostd/variant.h"
 // clang-format on
 
-#if defined(HAVE_ABSEIL) && !defined(HAVE_ABSEIL_VARIANT)
-
+#if defined(HAVE_ABSEIL)
```

Comment:
@owent - once this PR merged, could you please share some instructions, did we need this for MinGW? I think maybe we're not gonna be needing this code anymore. If we are building with `HAVE_ABSEIL` in future, we can demand that the implementation of `ThrowBadVariantAccess` comes from Abseil lib linked into the project. That way we can remove it from `core.cc` here.. Basically, I'm no longer using this exception handler for our own Abseil private variant, and most likely the code between lines 10 to 36 could be simply removed (in a separate PR 😄 ).